scipy/stats/_resampling.py

```diff
@@ -349,19 +353,60 @@ def bootstrap(data, statistic, *, n_resamples=9999, batch=None,
     >>> print(std_sample)
     3.9460644295563863
 
-    We can calculate a 90% confidence interval of the statistic using
-    `bootstrap`.
+    The bootstrap is used to approximate the variability we would expect if we
+    were to repeatedly sample from the unknown distribution and calculate the
+    statistic of the sample each time. It does this by repeatedly resampling
+    values *from the original sample* with replacement and calculating the
+    statistic of each resample. This results in a "bootstrap distribution" of
+    the statistic.
 
+    >>> import matplotlib.pyplot as plt
     >>> from scipy.stats import bootstrap
     >>> data = (data,)  # samples must be in a sequence
     >>> res = bootstrap(data, np.std, confidence_level=0.9,
     ...                 random_state=rng)
+    >>> plt.hist(res.bootstrap_distribution, bins=25)
+    >>> plt.title('Bootstrap Distribution')
+    >>> plt.xlabel('statistic value')
+    >>> plt.ylabel('frequency')
+    >>> plt.show()
+
+    The standard error quantifies this variability. It is calculated as the
+    standard deviation of the bootstrap distribution.
+
+    >>> res.standard_error
+    0.24427002125829136
+    >>> res.standard_error == np.std(res.bootstrap_distribution, ddof=1)
+    True
+
+    The bootstrap distribution of the statistic is often approximately normal
+    with scale equal to the standard error.
+
+    >>> x = np.linspace(3, 5)
+    >>> pdf = norm.pdf(x, loc=std_sample, scale=res.standard_error)
+    >>> plt.hist(res.bootstrap_distribution, bins=25, density=True)
+    >>> plt.plot(x, pdf)
+    >>> plt.title('Normal Approximation of the Bootstrap Distribution')
+    >>> plt.xlabel('statistic value')
+    >>> plt.ylabel('pdf')
+    >>> plt.show()
+
+    This suggests that we could construct a 90% confidence interval on the
+    statistic based on quantiles of this normal distribution.
+
+    >>> norm.interval(0.9, loc=std_sample, scale=res.standard_error)
+    (3.5442759991341726, 4.3478528599786)
+
+    However, the normal approximation is not always accurate. By default,
+    `bootstrap` uses more advanced techniques to make the confidence interval
+    more accurate.
```

Comment:
This is a bit coming out of nowhere. Can you add a brief sentence saying why it's not always accurate? Are you talking about multimodal distributions for instance?

Reply:
No. It's mostly because this can be used with arbitrary statistics, and not all statistics tend to produce normal bootstrap distributions. What would you suggest, then? Why were you expecting them to be normal - CLT?

Reply:
When reading the first sentence, I think it's not clear why we are saying a normal approximation is not always accurate. Because it's not expected to be normally distributed at all or because something else. This should be clearer.

Reply:
>  I think it's not clear why we are saying a normal approximation is not always accurate.

We are saying a normal approximation is not always accurate because the previous few blocks have been about generating a confidence interval using a normal approximation, but because this approximation is not always accurate, `bootstrap` uses advanced techniques for generate a more accurate confidence interval. 

It is important to state this because otherwise the reader might be confused about why the confidence interval generated by `bootstrap` is not the same as the confidence interval generated by the normal approximation.

I think it is helpful to mention the normal approximation in the first place, even thought `bootstrap` doesn't use it, because a) that is one way that the `standard_error` returned by `bootstrap` might be useful and b) many readers who have seen confidence intervals before have likely seen them in the context of a normal approximation of some sort. I want to tie this part about confidence intervals to their previous intuition, but I need to explain that `bootstrap` goes beyond that.

> Can you add a brief sentence saying why it's not always accurate? 

Perhaps the way to clarify this is to specifically address reasons why the reader might expect the normal approximation to be accurate? My only guess for that is the CLT, so I've added some detail to address that.